common/helper.py

```diff
@@ -280,7 +283,18 @@ def extract_archive(archive_path, extract_to):
         raise UnsupportedArchiveError(
             f"Unsupported archive extension {archive_path.suffix}")
 
-    package.extractall(extract_to)
+    members = None
+    if exclude and isinstance(exclude, list):
+        ignore = []
+        all_members = package.namelist()
+        for member in all_members:
+            for pattern in exclude:
+                if pattern in member:
+                    ignore.append(member)
+
+        members = [member for member in all_members if member not in ignore]
```

Comment:
it is better to create members list instead of creating not really used ignore list above